src/mapgen_flat.cpp

```diff
@@ -58,7 +58,10 @@ MapgenFlat::MapgenFlat(int mapgenid, MapgenParams *params, EmergeManager *emerge
 	//// amount of elements to skip for the next index
 	//// for noise/height/biome maps (not vmanip)
 	this->ystride = csize.X;
-	this->zstride = csize.X * (csize.Y + 2);
+	// 1-down overgeneration
+	this->zstride_1d = csize.X * (csize.Y + 1);
+	// 1-up 1-down overgeneration
+	this->zstride_1u1d = csize.X * (csize.Y + 2);
```

Comment:
In some mapgens only one type of zstride is used, but i kept both as i thought that good practice because until recently zstride was present but not used in mgv7.
However agreed i will remove the unused variables.